### PR TITLE
Appgallery

### PR DESCRIPTION
**Pull request by codemagic-mohammed**

### CODE REVIEW

File: content/yaml-publishing/huawei-appgallery.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+title: Huawei AppGallery
+description: How to deploy an app to Huawei AppGallery using codemagic.yaml
+weight: 4
+---
+
+Codemagic enables you to automatically publish your iOS or Android app to Huawei AppGallery.
+
+## Creating an API Client
```
Comment (nevercode-kevin) [inline content/yaml-publishing/huawei-appgallery.md:9]:
```suggestion
## Creating an API Client in Huawei AppGallery
```

File: content/yaml-publishing/huawei-appgallery.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+{{</notebox>}}
+
+4. After the client is successfully created, you need to save the **Client ID** and the **Client Secret** (**Key**) to [environment variables](/variables/environment-variable-groups/#storing-sensitive-valuesfiles) in a group named **app_gallery** for example.
+
+## Get the App ID
+1. Sign in to [AppGallery Connect](https://developer.huawei.com/consumer/en/service/josp/agc/index.html) and click **My apps** and select you app.
+3. Go to **App Information** and save the **App ID** to the environment variables as well in the **app_gallery** environment group.
+
+## App Signing
+If you want to release your app in App Bundle format and use the dynamic delivery feature, this service is required.
```
Comment (nevercode-kevin) [inline content/yaml-publishing/huawei-appgallery.md:27]: This sentence is ambiguous. Do you mean that the app will have to be signed if you want to release the app?

File: content/yaml-publishing/huawei-appgallery.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+First you need to sing you android application and upload the key to the Codemagic environment variables so Codemagic can sign you app before publishing it, see the [docs](https://docs.codemagic.io/yaml-code-signing/signing-android/).
```
Comment (nevercode-kevin) [inline content/yaml-publishing/huawei-appgallery.md:29]:
```suggestion
First, you need to sign your android application and upload the key to the Codemagic environment variables so Codemagic can sign your app before publishing it, see the [docs](https://docs.codemagic.io/yaml-code-signing/signing-android/).
```

File: content/yaml-publishing/huawei-appgallery.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+1. Go to your app page.
+2. Go to **App Signing** tab at the Services *section*.
```
Comment (nevercode-kevin) [inline content/yaml-publishing/huawei-appgallery.md:32]:
```suggestion
2. Go to **App Signing** tab in the *Services* section.
```

File: content/yaml-publishing/huawei-appgallery.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+3. Choose the first option *(Let AppGallery connect create and manage app signature for you)*.
+4. After you create your upload key, you need to export a PEM certificate for the new upload key.
+
+{{< highlight bash "style=paraiso-dark">}}
+$ keytool -export -rfc -keystore YOUR_KEYSTORE.jks -alias upload -file upload_certificate.pem
+{{< /highlight >}}
+
+5. Upload the **upload_certificate.pem** file and click **Submit**.
+
+## Adding the fastlane script to your code
```
Comment (nevercode-kevin) [inline content/yaml-publishing/huawei-appgallery.md:42]:
```suggestion
## Adding the Fastlane script to your code
```

File: content/yaml-publishing/huawei-appgallery.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+First you need to set up fastlane at your project by running `fastlane init` command, see the [docs](https://docs.fastlane.tools/getting-started/android/setup/).
```
Comment (nevercode-kevin) [inline content/yaml-publishing/huawei-appgallery.md:43]:
```suggestion
First, you need to set up Fastlane at your project by running `fastlane init` command, see the [docs](https://docs.fastlane.tools/getting-started/android/setup/).
```

**Review by nevercode-kevin — CHANGES_REQUESTED** [review]

File: content/yaml-publishing/huawei-appgallery.md (hunk 2 above)
Comment (codemagic-mohammed) [reply content/yaml-publishing/huawei-appgallery.md:27]: Yes, when publishing an `aab`.

File: content/yaml-publishing/huawei-appgallery.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+4. After the client is successfully created, you need to save the **Client ID** and the **Client Secret** (**Key**) to [environment variables](/variables/environment-variable-groups/#storing-sensitive-valuesfiles) in a group named **app_gallery** for example.
+
+## Get the App ID
+1. Sign in to [AppGallery Connect](https://developer.huawei.com/consumer/en/service/josp/agc/index.html) and click **My apps** and select you app.
+3. Go to **App Information** and save the **App ID** to the environment variables as well in the **app_gallery** environment group.
+
+## App Signing
+If you want to release your app in App Bundle format `.aab` and use the dynamic delivery feature, this service is required.
+
+First, you need to sign your android application and upload the key to the Codemagic environment variables so Codemagic can sign your app before publishing it, see the [docs](https://docs.codemagic.io/yaml-code-signing/signing-android/).
```
Comment (nevercode-kevin) [inline content/yaml-publishing/huawei-appgallery.md:29]:
```suggestion
If you want to release your app in App Bundle format `.aab` and use the dynamic delivery feature, you will need to sign your app. 

You will need to upload the signing key to the Codemagic environment variables so Codemagic can sign your app before publishing it, see the [docs](https://docs.codemagic.io/yaml-code-signing/signing-android/).
```

**Review by nevercode-kevin — CHANGES_REQUESTED** [review]

**Review by nevercode-kevin — APPROVED** [review]
LGTM

**Pull request merged**